File: Cora/Objective-C/Managers/UserDefaultsManager.m
#import "UserDefaultsManager.h"


@implementation UserDefaultsManager


+ (UserDefaultsManager *)sharedInstance {

	static UserDefaultsManager *sharedInstance = nil;
	static dispatch_once_t onceToken;

	dispatch_once(&onceToken, ^{

		sharedInstance = [self new];

	});

	return sharedInstance;

}


- (void)loadSwitchState {

	self.switchState = [NSUserDefaults.standardUserDefaults boolForKey:@"switchState"];

}


- (void)setSwitchState {

	[commandSwitch setOn:[NSUserDefaults.standardUserDefaults boolForKey:@"switchState"]];

}


- (void)saveSwitchState {

	[NSUserDefaults.standardUserDefaults setBool:commandSwitch.isOn forKey:@"switchState"];

}


- (void)loadAccentColor {

	NSData *decodedData = [NSUserDefaults.standardUserDefaults objectForKey:@"accentColor"];
	[ColorManager sharedInstance].accentColor = [NSKeyedUnarchiver unarchivedObjectOfClass:[UIColor class] fromData:decodedData error:nil];

}


- (void)saveAccentColor {

	NSData *encodedData = [NSKeyedArchiver archivedDataWithRootObject: [ColorManager sharedInstance].accentColor requiringSecureCoding:NO error:nil];
	[NSUserDefaults.standardUserDefaults setObject:encodedData forKey:@"accentColor"];

}


@end

File: Released/Cora/UIKit/Managers/UserDefaultsManager.swift
import UIKit


final class UserDefaultsManager {

	static let sharedInstance = UserDefaultsManager()
	var switchState = false

	private init() {}

 	func loadSwitchState() {
		switchState = UserDefaults.standard.bool(forKey: "switchState")
	}

	func setSwitchState() {
		GlobalManager.sharedInstance.commandSwitch.setOn(UserDefaults.standard.bool(forKey: "switchState"), animated: true)
	}

	func saveSwitchState() {
		UserDefaults.standard.set(GlobalManager.sharedInstance.commandSwitch.isOn, forKey: "switchState")
	}

	func loadAccentColor() {
		do {
			let decodedData = UserDefaults.standard.object(forKey: "accentColor")
			guard let data = decodedData as? Data else { return }

			guard let unarchivedColor = try NSKeyedUnarchiver.unarchivedObject(ofClass: UIColor.self, from: data) else {
				return
			}

			ColorManager.sharedInstance.accentColor = unarchivedColor
		}
		catch { return }

	}

	func saveAccentColor() {
		do {
			let encodedData = try NSKeyedArchiver.archivedData(withRootObject: ColorManager.sharedInstance.accentColor, requiringSecureCoding: false)
			UserDefaults.standard.set(encodedData, forKey: "accentColor")
		}
		catch { return }
	}

}
